frontend/src/components/SystemSubmitDrawer/index.tsx

```diff
@@ -40,6 +41,71 @@ enum State {
   other,
 }
 
+function hasCommonString(one: string, other: string): boolean {
+  return isSubstring(one, other) || isSubstring(other, one);
+}
+function isSubstring(query: string, string: string): boolean {
+  return string.toLowerCase().includes(query.toLocaleLowerCase());
+}
+
+type LangScore = {
```

Comment:
Both `type` and `interface` can be used for `LangScore` but I think it's usually better to use `interface` for something like this because interfaces can be merged, extended, and implemented so it's a lot more versatile. Here's a [blog post](https://blog.logrocket.com/types-vs-interfaces-in-typescript/) that goes more into the details. Also, could you please turn the comment into a docstring?